### PR TITLE
[FIX] purchase: tax inactives not shown on search

### PR DESCRIPTION
**Pull request by Ephoniz**
Issue: Currently if we have taxes for our Purchase no matter if we set the tax to inactive or we archive it, we will have access to it on the purchase_order_line.

Steps to reproduce:

- Install Purchase
- Create a new Tax and set it to inactive.
- Now create a new RfQ and in the lines add any product.
- Try to change the tax for this product.

Solution: since commit 8ff6747 we got rid of an odd domain which was allowing us to always get every tax no matter if they were active or not. In order for this domain to work we needed to set `context={'active_test': False}` which we no longer need and It's creating a bad behavior on how we want the active field on tax to act.

opw-3776871

### CODE REVIEW

**Review by HANNICHE-Walid — APPROVED** [review]
LGTM

**Comment by Whenrow** [comment]
@Ephoniz Seems that the context is still usefull alone https://github.com/odoo/odoo/commit/1f24bd28aa68dbe795ec8a99b07381790c1b4441

**Comment by HANNICHE-Walid** [comment]
@Whenrow the context was introduced in https://github.com/odoo/odoo/pull/147825
and the test that cover that use case (test_archived_tax_totals) now still passes without it

**Comment by Whenrow** [comment]
@HANNICHE-Walid Yes, I can see the runbot is ok, but as this [commit](https://github.com/odoo/odoo/commit/1f24bd28aa68dbe795ec8a99b07381790c1b4441) on sales order says, the invoice could have different taxes than the purchase order if we archive the tax before the bill creation

**Comment by Ephoniz** [comment]
@Whenrow Thanks for the review! Indeed we still need it for the case where we archive the Tax after we have already done requests for quotations with it, if we remove the context then after archive we no longer see the tax in those past RfQ which will be weird since we still can see it in the bill. IMO, we can just add the domain to the view where it should be.

**Comment by Whenrow** [comment]
@Ephoniz Yes. it's the best solution to me too

**Comment by Ephoniz** [comment]
@Whenrow Nice, I don't know If you can approve this PR or should I wait for the team approval :)

**Comment by Whenrow** [comment]
robodoo r+
@Ephoniz Thanks

**Pull request closed without merging**
